src/ObjCRuntime/Runtime.cs

```diff
@@ -158,6 +161,15 @@ internal unsafe struct InitializationOptions {
 					return (Flags & InitializationFlags.IsSimulator) == InitializationFlags.IsSimulator;
 				}
 			}
+
+#if NET
+			// Future optimization potential: make the linker change this into a constant.
```

Comment:
better file an issue :)

Reply:
> better file an issue :)

Here it is: https://github.com/xamarin/xamarin-macios/issues/10903

Reply:
CoreCLR would be able to optimize readonly static with initilizer btw.

Reply:
> CoreCLR would be able to optimize readonly static with initilizer btw.

That would require a refactoring to do the initialization from a static constructor (which is not impossible, just a bit more work).

Also that does not make the linker able to remove additional dependencies that results from constant propagation:

```csharp
if (IsCoreCLR) {
    DoThisThing ();
} else {
    DoThisOtherThing ();
}
```

using the linker to replace IsCoreCLR with a constant value, also makes the linker able to remove the call to either `DoThisThing` or `DoThisOtherThing`, along with the corresponding implementation, and so on, and this can get quite big.

Reply:
Sure, it makes sense to link it out as a constant for the final deployment. The `static readonly` would however work even for the development scenario without linker. The JIT would be able to eliminate the other branch just like linker can.